### PR TITLE
[SDP-774] Update tx worker to use crash tracker on failed tx's

### PR DESCRIPTION
**Pull request by ziyliu**
### What
- Moved horizon tx error parsing logic to utils file
- Call crash tracker when tx should be marked as an error

### Why

[TODO: Why this change is being made. Include any context required to understand the why.]

### Known limitations

[TODO or N/A]

#### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).
* [x] This PR's title starts with the name of the package, area, or subject affected by the change.

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [ ] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [ ] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

**Review by CaioTeixeira95 — APPROVED** [review]
Awesome! 🚀

File: internal/transactionsubmission/utils/errors.go [hunk 1]
```diff
@@ -277,6 +277,38 @@ func (e *HorizonErrorWrapper) IsDestinationAccountNotReady() bool {
 		e.IsLineFull())
 }
 
+// ShouldMarkAsError determines whether a transaction neeeds to be marked as an error based on the
+// transaction error code or failed op code so that TSS can determine whether it needs
+// to be retried.
+func (e *HorizonErrorWrapper) ShouldMarkAsError() bool {
```
Comment (marcelosalloum) [inline internal/transactionsubmission/utils/errors.go:283]: 👋 I'm missing a unit test for this method.

**Review by marcelosalloum — APPROVED** [review]
LGTM! Plz, feel free to merge it after addressing the suggestion below.

Also, do you think [these comments](https://github.com/stellar/stellar-disbursement-platform-backend/blob/46428417d3e1f46899dad4f075850a4e0e75092d/internal/transactionsubmission/transaction_worker.go#L183-L221) are still valuable? Or can we delete them?

**Comment by ziyliu** [comment]
> LGTM! Plz, feel free to merge it after addressing the suggestion below.
> 
> Also, do you think [these comments](https://github.com/stellar/stellar-disbursement-platform-backend/blob/46428417d3e1f46899dad4f075850a4e0e75092d/internal/transactionsubmission/transaction_worker.go#L183-L221) are still valuable? Or can we delete them?

let's leave them in there, I think it could be helpful to other devs who want to touch this code 🙃

**Pull request merged**